pennylane/devices/default_mixed.py

```diff
@@ -903,5 +977,95 @@ def execute(
+
+        Args:
+            execution_config (ExecutionConfig): an unprocessed execution config.
+
+        Returns:
+            ExecutionConfig: a preprocessed execution config.
+        """
+        updated_values = {}
+        for option in execution_config.device_options:
+            if option not in self._device_options:
+                raise qml.DeviceError(f"device option {option} not present on {self}")
+
+        if execution_config.gradient_method == "best":
+            updated_values["gradient_method"] = "backprop"
+        updated_values["use_device_gradient"] = execution_config.gradient_method in {
+            "backprop",
+            "best",
+        }
+        updated_values["grad_on_execution"] = False
+        updated_values["device_options"] = dict(execution_config.device_options)  # copy
+
+        for option in self._device_options:
+            if option not in updated_values["device_options"]:
+                updated_values["device_options"][option] = getattr(self, f"_{option}")
+        return replace(execution_config, **updated_values)
+
+    @debug_logger
+    def preprocess(
+        self,
+        execution_config: ExecutionConfig = DefaultExecutionConfig,
+    ) -> tuple[TransformProgram, ExecutionConfig]:
+        """This function defines the device transform program to be applied and an updated device
+        configuration.
+
+        Args:
+            execution_config (Union[ExecutionConfig, Sequence[ExecutionConfig]]): A data structure
+                describing the parameters needed to fully describe the execution.
+
+        Returns:
+            TransformProgram, ExecutionConfig: A transform program that when called returns
+            ``QuantumTape`` objects that the device can natively execute, as well as a postprocessing
+            function to be called after execution, and a configuration with unset
+            specifications filled in.
+
+        This device:
+
+        * Supports any qubit operations that provide a matrix
+        * Supports any qubit channel that provides Kraus matrices
+
+        """
+        config = self._setup_execution_config(execution_config)
+        transform_program = TransformProgram()
+
+        transform_program.add_transform(validate_device_wires, self.wires, name=self.name)
+        transform_program.add_transform(
+            decompose,
+            stopping_condition=stopping_condition,
+            name=self.name,
+        )
+        transform_program.add_transform(validate_measurements, name=self.name)
+        transform_program.add_transform(
+            validate_observables, stopping_condition=observable_stopping_condition, name=self.name
+        )
+
+        if config.gradient_method == "backprop":
+            transform_program.add_transform(no_sampling, name="backprop + default.mixed")
+
+        if self.readout_err is not None:
+            transform_program.add_transform(warn_readout_error_state)
```

Comment:
This will probably never bother anyone, but if a user has it at `None` and later updates the device with `dev.readout_err = 0.1`, this transform won't be added, and we'll miss this validation step. It seems like `redout_err` is meant to be set on initialization and then not updated. Can we make it a non-settable property? This could be done by storing it as `self._readout_err` in the `__init__`, so its private and people know they aren't supposed to touch it, and then adding a:

```
@property
def redout_err(self):
    return self._readout_err
```
    
to the device. You'll see similar things on a lot of other devices to try to make it clearer to users what things are attributes they can update on a device, and which things might act strangely if they try to use them.

Alternatively, if the old device _did_ treat this as a property users could update on the device, maybe we need to include the check in the transform instead, and include the transform unconditionally.

Reply:
Good catch and I can totally get its importance. Right now all the implementations are based on qutrit mixed dev to minimize risks so I guess we better have a separate story to have this idea as an improvement for both qubit mixed and qutrit mixed in the future.

Reply:
Backlogged here [sc-79441]